lib/views/reviews-view.js

```diff
@@ -0,0 +1,573 @@
+import path from 'path';
+import React, {Fragment} from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import {EnableableOperationPropType} from '../prop-types';
+import Tooltip from '../atom/tooltip';
+import Commands, {Command} from '../atom/commands';
+import AtomTextEditor from '../atom/atom-text-editor';
+import {getDataFromGithubUrl} from './issueish-link';
+import EmojiReactionsController from '../controllers/emoji-reactions-controller';
+import {checkoutStates} from '../controllers/pr-checkout-controller';
+import GithubDotcomMarkdown from './github-dotcom-markdown';
+import PatchPreviewView from './patch-preview-view';
+import CheckoutButton from './checkout-button';
+import Timeago from './timeago';
+import Octicon from '../atom/octicon';
+import RefHolder from '../models/ref-holder';
+import {toNativePathSep} from '../helpers';
+import {addEvent} from '../reporter-proxy';
+
+export default class ReviewsView extends React.Component {
+  static propTypes = {
+    // Relay results
+    relay: PropTypes.shape({
+      environment: PropTypes.object.isRequired,
+    }).isRequired,
+    repository: PropTypes.object.isRequired,
+    pullRequest: PropTypes.object.isRequired,
+    summaries: PropTypes.array.isRequired,
+    commentThreads: PropTypes.arrayOf(PropTypes.shape({
+      thread: PropTypes.object.isRequired,
+      comments: PropTypes.arrayOf(PropTypes.object).isRequired,
+    })),
+    refetch: PropTypes.func.isRequired,
+
+    // Package models
+    multiFilePatch: PropTypes.object.isRequired,
+    contextLines: PropTypes.number.isRequired,
+    checkoutOp: EnableableOperationPropType.isRequired,
+    summarySectionOpen: PropTypes.bool.isRequired,
+    commentSectionOpen: PropTypes.bool.isRequired,
+    threadIDsOpen: PropTypes.shape({
+      has: PropTypes.func.isRequired,
+    }),
+    postingToThreadID: PropTypes.string,
+    scrollToThreadID: PropTypes.string,
+    commentTranslations: PropTypes.object,
```

Comment:
I wish there was a `PropTypes.map` so we could be more specific about what this prop actually is, just for the sake of readability.  Maybe it's worth making a custom PropTypes validator.

Reply:
I'm "ehhhh" on the custom validator idea - it sounds like a fair amount of effort for not much payoff. Since it's only for readability I'm just going to add a comment here.

Also the real, _real_ answer for this would be "use TypeScript" :smile: